spacy/matcher/matcher.pyx

```diff
@@ -215,16 +215,17 @@ cdef class Matcher:
         else:
             raise ValueError(Errors.E195.format(good="Doc or Span", got=type(doclike).__name__))
         cdef Pool tmp_pool = Pool()
-        if TAG in self._seen_attrs and not doc.has_annotation("TAG"):
-            raise ValueError(Errors.E155.format(pipe="tagger", attr="TAG"))
-        if POS in self._seen_attrs and not doc.has_annotation("POS"):
-            raise ValueError(Errors.E155.format(pipe="morphologizer", attr="POS"))
-        if MORPH in self._seen_attrs and not doc.has_annotation("MORPH"):
-            raise ValueError(Errors.E155.format(pipe="morphologizer", attr="MORPH"))
-        if LEMMA in self._seen_attrs and not doc.has_annotation("LEMMA"):
-            raise ValueError(Errors.E155.format(pipe="lemmatizer", attr="LEMMA"))
-        if DEP in self._seen_attrs and not doc.has_annotation("DEP"):
-            raise ValueError(Errors.E156.format())
+        if not suppress_errors:
+            if TAG in self._seen_attrs and not doc.has_annotation("TAG"):
+                raise ValueError(Errors.E155.format(pipe="tagger", attr="TAG"))
+            if POS in self._seen_attrs and not doc.has_annotation("POS"):
+                raise ValueError(Errors.E155.format(pipe="morphologizer", attr="POS"))
+            if MORPH in self._seen_attrs and not doc.has_annotation("MORPH"):
+                raise ValueError(Errors.E155.format(pipe="morphologizer", attr="MORPH"))
+            if LEMMA in self._seen_attrs and not doc.has_annotation("LEMMA"):
+                raise ValueError(Errors.E155.format(pipe="lemmatizer", attr="LEMMA"))
+            if DEP in self._seen_attrs and not doc.has_annotation("DEP"):
+                raise ValueError(Errors.E156.format())
```

Comment:
We should probably make this a loop:

```
for attr in [TAG, POS, MORPH, LEMMA, DEP]:
    if attr in self._seen_attrs and not doc.has_annotation(attr):
        raise ValueError(Errors.E155.format(pipe="morphologizer", attr=self.strings[attr]))
```

Reply:
Yeah. (This is from when the is_flags weren't easily derivable from the features and they were the same for MORPH/POS.)